### ISSUE TITLE
neg images collect

### ISSUE THREAD

**Issue by ghost** [mks0601/Face_detection#1]
thanks for your share, and how many neg images in your model,and did you collect neg images randomly?or other method?

**Comment by mks0601** [comment]
I followed the method of the paper.
for 12-net, neg images are randomly cropped with various size, scale, position
for 24, 48-net, i used conventional hard neg mining method according to the paper

Best,

## Gyeongsik Moon

Gyeongsik Moon
M.S. Candidate
Department of ECE, SNU, Seoul, Korea
http://cv.snu.ac.kr/ http://cv.snu.ac.kr/hmyeong/

**Comment by ghost** [comment]
how many negative images in your train demo? as many as possible?

**Comment by mks0601** [comment]
## for 24,48-net, as many as possible

Gyeongsik Moon
M.S. Candidate
Department of ECE, SNU, Seoul, Korea
http://cv.snu.ac.kr/ http://cv.snu.ac.kr/hmyeong/

**Comment by melgor** [comment]
Hi,
I am trying to reproduce your results on FDDB. I have a question regarding training the 24-net and 48-net.
First of all, I learn 12-net on ALFW (50k positives with flipes, negative 0.5M). This net works ok on FDDB, reach 0.9 on 350k false-posive,

Then, according to your test, I run hard-negative mining at images (exacly ImageNet validation set, 50k images). Here I collected 2.5M  negative examples. After training 100 epochs like in your code, this net it not able to detect any other face different from training set. The score at FDDB is very low (0.2 on 14k negatives, with threshold = 0.1). 

Analyzing it suggest that Net overfitt after 100 epoch of training data (especially Positive Data).

Do you have same situation? 
How long did you train your nets, for 100 like presented at code?
What negative images did you use?
What are the threshold level for each net (12 and 24). This in you code are really low, so it look like resolving same task (binary classification)  by each net independently, not using cascade structure. Or I am wrong?

**Comment by mks0601** [comment]
Hi.
First of all, I followed the methods of paper entirely.

So, when I collect neg patches to train my 12-net, i used 200,000non face patches unlike yours(0.5M)
Anyway, after you train the 12-net, you have to make the recall rate and # of window like the table1. It`s quite a long time ago that i implemented this paper, but i remember my thresholds were very small(1e-2~1e-4).

I used coco dataset for neg db.

## 

Gyeongsik Moon
M.S. Candidate
Department of ECE, SNU, Seoul, Korea
http://cv.snu.ac.kr/ http://cv.snu.ac.kr/hmyeong/

**Comment by asdfghwunai** [comment]
Hi.Thanks for your great work.I can‘t understand how to adjust the threshold.I used your 12net's threshold to run the hard negative mining and finally the 12 net's db size is five millions.This is too large.Can you tell me the details how  you get the three thresholds？
